article.md

```diff
@@ -0,0 +1,322 @@
+## Background
+Make sure that you're familiar with [cairo-lang](https://www.cairo-lang.org) as it is essential in this case.
+There is a whole dedicated [piece](https://blog.swmansion.com/testing-starknet-contracts-made-easy-with-protostar-2ecdad3c9133) explaining almost everything you need to know about contracts on StarkNet.
+# Use case - creating a binding between ETH address and Starknet Address
+
+## Message structure
+```json
+{
+  "domain": {
+    "name": "Confirm the ownership of this StarkNet address",
+    "contractAddress": "0xFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF",
+    "version": "1"
+  },
+
+  "primaryType": "Payload",
+  "types": {
+    "domain": [
+      { "name": "name", "type": "string" },
+      { "name": "contractAddress", "type": "address"},
+      { "name": "version", "type": "string" }
+    ],
+    "Payload": [
+      { "name": "starknetAddress", "type": "uint256" }
+    ]
+  }
+}
+```
+This is the structure that we'll be using in this example. Note that `domain` here represents `eip712Domain` which, after hashing, will become a domain separator that we will need.<br/>
+Elements of the domain are as follows:
+* `name` - the user readable name of signing domain, i.e. the name of the DApp or the protocol.
+* `version` - the current major version of the signing domain. Signatures of messages using different versions are not compatible.
+* `contractAddress` - the address of the contract that will verify the signature. It ensures that the signature can only be used by our contract on the given chain.
+
+`Payload` will be the main structure that we'll be constructing and signing. For the sake of simplicity, it only has one field but the EIP-712 standard offers a lot of flexibility.
+
+
+## Signing with an Ethereum wallet
+Let's use Metamask as an example. You just need to call the `signTypedData_v4` ffunction with proper parameters and it will return the signature that we need.<br/>
+```js
+const msg = {
+  domain: {
+    name: "Confirm the ownership of this StarkNet address",
+    verifyingContract: '0xFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF',
+    version: '1',
+  },
+  message: {
+    "starknetAddress": 12356789,
+  },
+  primaryType: 'Payload',
+  types: {
+    domain: [
+      { name: 'name', type: 'string' },
+      { name: 'verifyingContract', type: 'address' },
+      { name: 'version', type: 'string' },
+    ],
+    Payload: [
+      { name: 'starknetAddress', type: 'uint256' },
+    ],
+  },
+};
+const accounts = await window.ethereum.request({method: "eth_requestAccounts"})
+await window.ethereum.request({
+    method: "eth_signTypedData_v4",
+    from: accounts[0],
+    params: [accounts[0], JSON.stringify(msg)]
+})
+```
+After you run this this JS code in the browser's console on the MetaMask website it will trigger a signing request. An application would use a similar code to get user's accounts and signature. <br/><br/>
```

Comment:
"...run this this JS code in the browser's console with MetaMask enabled it will trigger..."